vms/platformvm/state/disk_staker_diff_iterator.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package state
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+)
+
+const (
+	// diffKey = [subnetID] + [inverseHeight] + [nodeID]
+	startDiffKeyLength  = ids.IDLen + database.Uint64Size
+	diffKeyLength       = startDiffKeyLength + ids.NodeIDLen
+	diffKeyHeightOffset = ids.IDLen
+	diffKeyNodeIDOffset = diffKeyHeightOffset + database.Uint64Size
+
+	// weightValue = [isNegative] + [weight]
+	weightValueLength = 1 + database.Uint64Size
+	weightValueTrue   = 0x01
+)
+
+var (
+	errUnexpectedDiffKeyLength     = fmt.Errorf("expected diff key length %d", diffKeyLength)
+	errUnexpectedWeightValueLength = fmt.Errorf("expected weight value length %d", weightValueLength)
+)
+
+// getStartDiffKey is used to determine the starting key when iterating.
+//
+// Note: the result should be a prefix of [getDiffKey] if called with the same
+// arguments.
+func getStartDiffKey(subnetID ids.ID, height uint64) []byte {
+	key := make([]byte, startDiffKeyLength)
+	copy(key, subnetID[:])
+	packIterableHeight(key[diffKeyHeightOffset:], height)
+	return key
+}
+
+func getDiffKey(subnetID ids.ID, height uint64, nodeID ids.NodeID) []byte {
+	key := make([]byte, diffKeyLength)
+	copy(key, subnetID[:])
+	packIterableHeight(key[diffKeyHeightOffset:], height)
+	copy(key[diffKeyNodeIDOffset:], nodeID[:])
+	return key
+}
+
+func parseDiffKey(key []byte) (ids.ID, uint64, ids.NodeID, error) {
+	if len(key) != diffKeyLength {
+		return ids.Empty, 0, ids.EmptyNodeID, errUnexpectedDiffKeyLength
+	}
+	var (
+		subnetID ids.ID
+		nodeID   ids.NodeID
+	)
+	copy(subnetID[:], key)
```

Comment:
Didn't actually know you could limit the amount copied by providing a smaller `dst`. Confirmed this is correct.